diff --git a/node_modules/tss-react/esm/tools/classnames.js b/node_modules/tss-react/esm/tools/classnames.js
index 506f2c0..4f7282d 100644
--- a/node_modules/tss-react/esm/tools/classnames.js
+++ b/node_modules/tss-react/esm/tools/classnames.js
@@ -1,5 +1,14 @@
 import { assert } from "./assert";
 import { typeGuard } from "./typeGuard";
+
+function hasCustomToString(arg) {
+  return (
+    typeof arg.toString === "function" &&
+    arg.toString !== Object.prototype.toString &&
+    !String(arg.toString).includes("[native code]")
+  );
+}
+
 /** Copy pasted from
  * https://github.com/emotion-js/emotion/blob/23f43ab9f24d44219b0b007a00f4ac681fe8712e/packages/react/src/class-names.js#L17-L63
  **/
@@ -21,6 +30,15 @@ export const classnames = (args) => {
                 }
                 else {
                     assert(!typeGuard(arg, false));
+
+                    // This patch handles legacy glamor style objects that are passed to "cx" function
+                    // and have not been invoked with toString. Without the pathh those classes
+                    // would not be added to the class list.
+                    if (hasCustomToString(arg)) {
+                      toAdd = arg.toString();
+                      break;
+                    }
+
                     if (process.env.NODE_ENV !== "production" &&
                         arg.styles !== undefined &&
                         arg.name !== undefined) {
