diff --git a/node_modules/swiper/shared/utils.mjs b/node_modules/swiper/shared/utils.mjs
index 87bf93e..4cb726a 100644
--- a/node_modules/swiper/shared/utils.mjs
+++ b/node_modules/swiper/shared/utils.mjs
@@ -185,7 +185,7 @@ function elementChildren(element, selector) {
     selector = '';
   }
   const window = getWindow();
-  const children = [...element.children];
+  const children = Array.from(element.children);
   if (window.HTMLSlotElement && element instanceof HTMLSlotElement) {
     children.push(...element.assignedElements());
   }
@@ -202,7 +202,11 @@ function elementIsChildOfSlot(el, slot) {
     if (el === elementToCheck) {
       return true;
     }
-    elementsQueue.push(...elementToCheck.children, ...(elementToCheck.shadowRoot?.children || []), ...(elementToCheck.assignedElements?.() || []));
+    elementsQueue.push(
+      ...elementToCheck.children,
+      ...(elementToCheck.shadowRoot && elementToCheck.shadowRoot.children ? elementToCheck.shadowRoot.children : []),
+      ...(elementToCheck.assignedElements ? elementToCheck.assignedElements() : [])
+    );
   }
 }
 function elementIsChildOf(el, parent) {
